src/cli/init.rs

```diff
@@ -70,28 +94,64 @@ pub async fn execute(args: Args) -> miette::Result<()> {
     // Fail silently if it already exists or cannot be created.
     fs::create_dir_all(&dir).ok();
 
-    // Write pixi.toml
-    let name = dir
-        .file_name()
-        .ok_or_else(|| {
-            miette::miette!(
-                "Cannot get file or directory name from the path: {}",
-                dir.to_string_lossy()
-            )
-        })?
-        .to_string_lossy();
-    let version = "0.1.0";
-    let author = get_default_author();
-    let channels = if let Some(channels) = args.channels {
-        channels
+    let (name, channels, conda_deps, pip_deps) = if let Some(env_file) = args.env_file {
+        let env_info = read_env_yml(env_file)?;
+        let name = env_info.name;
+        let channels = env_info.channels;
+
+        let mut conda_deps = vec![];
+        let mut pip_deps = vec![];
+
+        for dep in env_info.dependencies {
+            match dep {
+                CondaEnvDep::Conda(d) => {
+                    conda_deps.push(MatchSpec::from_str(&d).into_diagnostic()?)
+                }
+                CondaEnvDep::Pip { pip } => pip_deps.extend(
+                    pip.into_iter()
+                        .map(|d| {
+                            let req = pep508_rs::Requirement::from_str(&d).into_diagnostic()?;
+                            let name = rip::types::PackageName::from_str(req.name.as_str())?;
+                            let requirement = PyPiRequirement::from(req);
+                            Ok((name, requirement))
+                        })
+                        .collect::<miette::Result<Vec<_>>>()?,
+                ),
+            }
+        }
+
+        if !pip_deps.is_empty() {
+            conda_deps.push(MatchSpec::from_str("pip").into_diagnostic()?);
+        }
+
+        (name, channels, conda_deps, pip_deps)
```

Comment:
Could you move this into a function and add some tests where we can input env.ymls and review the outcome?

Reply:
I've moved the dependency parsing logic into a seperate function and added a test case to test this specific functionality. Do let me know if anything more is required.